### PR TITLE
Make dropdown content alignable

### PR DESCRIPTION
**Pull request by devkanro**
Change dropdown content container from Row to Box, and users can use `Modifier.align` to align content

### CODE REVIEW

File: core/src/main/kotlin/org/jetbrains/jewel/Dropdown.kt [hunk 1]
```diff
@@ -116,25 +118,28 @@ fun Dropdown(
...
-                    verticalAlignment = Alignment.CenterVertically,
-                    content = {
-                        content()
-                    },
+                Box(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(style.metrics.contentPadding)
+                        .padding(
+                            end = minSize.height - style.metrics.contentPadding.calculateRightPadding(
```
[the remark this answers is not in the document]
Comment (devkanro) [reply core/src/main/kotlin/org/jetbrains/jewel/Dropdown.kt:133]: In fact, the best way is to provide a custom layout for it.

[the remark this answers is not in the document]
Comment (devkanro) [reply core/src/main/kotlin/org/jetbrains/jewel/Dropdown.kt:133]: Also, is the icon on the left when Dropdown is RTL layout direction?

[the remark this answers is not in the document]
Comment (rock3r) [reply core/src/main/kotlin/org/jetbrains/jewel/Dropdown.kt:133]: Should be yeah

[the remark this answers is not in the document]
Comment (rock3r) [reply core/src/main/kotlin/org/jetbrains/jewel/Dropdown.kt:133]: Although to be clear in Swing there is no support for RTL whatsoever, and the specs don't have it either, so there is no reference

[the remark this answers is not in the document]
Comment (devkanro) [reply core/src/main/kotlin/org/jetbrains/jewel/Dropdown.kt]: 截屏2023-10-13 17 35 40

[the remark this answers is not in the document]
Comment (devkanro) [reply core/src/main/kotlin/org/jetbrains/jewel/Dropdown.kt]: Work fine with LTR

[the remark this answers is not in the document]
Comment (rock3r) [reply core/src/main/kotlin/org/jetbrains/jewel/Dropdown.kt]: Perfect, thanks for checking

**Review by rock3r — APPROVED** [review]

**Pull request merged**